From: John Feeney <jfeeney@redhat.com>
Date: Wed, 17 Feb 2010 20:34:11 -0500
Subject: [net] tg3: fix 57765 LED
Message-id: <4B7C52C3.5050001@redhat.com>
Patchwork-id: 23322
O-Subject: [RHEL5.5 PATCH] Fix tg3 57765 LED
Bugzilla: 566016
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Andy Gospodarek <gospo@redhat.com>

bz566016
https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=566016
tg3: 57765 LED does not work correctly

Description of problem:
Broadcom just recently posted this patch upstream that fix the
support of 57765. More specifically, the UART port on the 57765
interferes with the correct operation of the device's LEDs.

Solution:
If the debug UART is left enabled, the LEDs will not work properly.
This patch disables the debug UART.

Upstream status:
Matt Carlson, the Broadcom tg3 maintainer, posted the patch on
netdev 1/20/2010 (net-next 08/10 tg3: Turn off the debug UART for
57765).

Brew:
Successfully built in Brew for all architectures
(task 2263880).

Testing:
I have run Connectathon successfully with the x86_64
debug kernel on RHTS systems with the following tg3 NICs.
  5722, 5761, 5721, 5755, 5704, 5755, 5764, 5703a, 5721,
  5703x, 5764, 5715, 5704.

Acks would be appreciated. Thanks.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/tg3.c b/drivers/net/tg3.c
index ec3caad..85896ae 100644
--- a/drivers/net/tg3.c
+++ b/drivers/net/tg3.c
@@ -13355,7 +13355,8 @@ static int __devinit tg3_get_invariants(struct tg3 *tp)
 		tp->grc_local_ctrl |= GRC_LCLCTRL_GPIO_OE3;
 
 	if (GET_ASIC_REV(tp->pci_chip_rev_id) == ASIC_REV_5755 ||
-	    GET_ASIC_REV(tp->pci_chip_rev_id) == ASIC_REV_57780)
+	    GET_ASIC_REV(tp->pci_chip_rev_id) == ASIC_REV_57780 ||
+	    GET_ASIC_REV(tp->pci_chip_rev_id) == ASIC_REV_57765)
 		tp->grc_local_ctrl |= GRC_LCLCTRL_GPIO_UART_SEL;
 
 	if (tp->pdev->device == PCI_DEVICE_ID_TIGON3_5761 ||
